subsys/debug/Kconfig

```diff
@@ -163,6 +163,19 @@ config ASSERT_LEVEL
 	  Level 1: on + warning in every file that includes __assert.h
 	  Level 2: on + no warning
 
+config SPIN_VALIDATE
+	bool "Enable spinlock validation"
+	depends on ASSERT
+	default y if MP_NUM_CPUS < 4 && (!FLASH || FLASH_SIZE > 32)
+	help
+	  There's a spinlock validation framework available when asserts are
+	  enabled. It adds a relatively hefty overhead (about 3k or so) to
+	  kernel code size, don't use on platforms known to be small. (Note
+	  we're using the kconfig value here.  This isn't defined for every
+	  board, but the default of zero works well as an "infinity"
+	  fallback.  There is a DT_FLASH_SIZE parameter too, but that seems
+	  even more poorly supported.
```

Comment:
You can take out my nitpicky comment about kconfig vs. devicetree memory size detection here, it's not relevant anymore since this is 100% kconfig.

Reply:
Will do :-)